### ISSUE TITLE
Segmentation mask becomes small when moved closer to the object

### ISSUE THREAD

**Issue by ArghyaChatterjee** [ultralytics/ultralytics#24168]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, 

I am trying to use my yolov8 trained model on my custom dataset and I get the door panel segmentation being clipped when the robot moves closer to the door panel as if the mask resolution can not be bigger than this. What can be the solution for this one ?

https://github.com/user-attachments/assets/d038c92d-6735-4f28-bbde-99313f5aff96

Youtube Link: https://youtu.be/z7fllca9qb4

Here are my training parameters:
  "split": {
    "train_ratio": 0.9,
    "seed": 42
  },

  "training": {
    "model": "yolov8n-seg.pt",
    "epochs": 100,
    "img_size": 640,
    "batch_size": 16,
    "mask_ratio": 1,
    "scale": 0.9,
    "copy_paste": 0.3,
    "degrees": 10.0,
    "device": 0,
    "run_name_prefix": "multi_class_yolov8_reclassified",
    "exist_ok": true
  },

  "export": {
    "enabled": true,
    "format": "onnx",
    "imgsz": [720, 1280],
    "opset": 11,
    "save_subdir": "weights"
  }

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ArghyaChatterjee, thank you for your interest in Ultralytics 🚀! This is an automated response to help gather the details needed for review, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this appears to be a segmentation behavior question/possible bug with a custom-trained YOLO model, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate 🔍.
Please also include as much information as possible, such as:
- The exact `predict` or deployment/export workflow you are using
- Whether the issue appears in native inference before export, or only after ONNX export
- A sample image and its corresponding expected vs actual mask result
- Training logs and validation metrics
- Your full environment details and `ultralytics` version
- Confirmation that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you try using YOLO26-Seg? And your export `imgsz` is too large. It should ideally match your training `imgsz` for best accuracy.

**Comment by 1548093028** [comment]
This is likely caused by DFL (Distribution Focal Loss). When v8 predicts bounding boxes, it predicts ltrb (left, top, right, bottom) with mathematical expectations ranging from 0 to 15. The largest detection head has a stride of 32, so the maximum detection box width is (15+15) × 32 = 960. When the image resolution exceeds this value, this issue occurs.
To use larger resolutions for inference, you need to either increase the DFL value from the original 16 during training, or add a P6 detection head.

**Comment by glenn-jocher** [comment]
That looks consistent with box saturation, not a `scale_masks()` bug: the seg pipeline crops masks to the predicted box in [`crop_mask()`](https://docs.ultralytics.com/reference/utils/ops/#ultralytics.utils.ops.crop_mask), while [`scale_masks()`](https://docs.ultralytics.com/reference/utils/ops/#ultralytics.utils.ops.scale_masks) only rescales afterward, so if the box tops out at larger object sizes the mask will appear clipped too; please test the same clip at `imgsz=640`, and if that removes it the fix is to retrain for a larger `imgsz` rather than exporting a 640-trained model for 1280-wide inference.

**Comment by ArghyaChatterjee** [comment]
Thanks everyone. It looks like the problem is gone when I used "img_size": 960 but the training time increased significantly in that process. I tried with 1280 but looks like it's asking for more GPUs/VRAM to train with that image size. Probably the fix is to export in the same image size as the input during training.

**Comment by glenn-jocher** [comment]
Yes, that sounds right: keeping training and inference/export `imgsz` aligned is the safest path here, and if `1280` is too expensive then `960` is a reasonable compromise; another practical option is to tile or crop the input so close objects stay within the scale range seen during training.
